addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3380,9 +3380,21 @@ export class OdooEditor extends EventTarget {
         if (!rangeContent.hasChildNodes()) {
             return;
         }
+        // find list of ancestors.
+        const ancestorList = range.commonAncestorContainer.nodeType === Node.ELEMENT_NODE ?
+                            ancestors(range.commonAncestorContainer.firstChild) : ancestors(range.commonAncestorContainer);
+        const closestLi = closestElement(selection.anchorNode, 'li');
+        for (const ancestor of ancestorList) {
+            if ((closestLi && rangeContent.firstChild.nodeName === 'LI') || isBlock(rangeContent.firstChild)) {
+                break;
+            }
+            const clone = ancestor.cloneNode();
+            clone.append(...rangeContent.childNodes);
+            rangeContent.appendChild(clone);
+        }
```

Comment:
I feel like this is the really interesting part of the change. Now that I have understood this, I think the comment above `const ancestorsList` could be:
```js
        // Wrap rangeContent with clones of their ancestors to keep the styles.
```

Reply:
@dmo-odoo I have added this comment before `for loop` as I have added another comment for `const ancestorsList` that could be helpful.